packages/react/src/drafts/UnderlinePanels/UnderlinePanels.tsx

```diff
@@ -0,0 +1,193 @@
+import React, {Children, isValidElement, cloneElement, useState, useRef, type FC, type PropsWithChildren} from 'react'
+import {TabContainerElement} from '@github/tab-container-element'
+import {createComponent} from '../../utils/create-component'
+import {
+  StyledUnderlineItemList,
+  StyledUnderlineWrapper,
+  UnderlineItem,
+  type UnderlineItemProps,
+} from '../../internal/components/UnderlineTabbedInterface'
+import Box, {type BoxProps} from '../../Box'
+import {useId} from '../../hooks'
+import {invariant} from '../../utils/invariant'
+import type {IconProps} from '@primer/octicons-react'
+import {merge, type BetterSystemStyleObject, type SxProp} from '../../sx'
+import {defaultSxProp} from '../../utils/defaultSxProp'
+import {useResizeObserver, type ResizeObserverEntry} from '../../hooks/useResizeObserver'
+import useIsomorphicLayoutEffect from '../../utils/useIsomorphicLayoutEffect'
+
+export type UnderlinePanelsProps = {
+  /**
+   * Tabs (UnderlinePanels.Tab) and panels (UnderlinePanels.Panel) to render
+   */
+  children: React.ReactNode
+  /**
+   * Accessible name for the tab list
+   */
+  'aria-label'?: React.AriaAttributes['aria-label']
+  /**
+   * ID of the element containing the name for the tab list
+   */
+  'aria-labelledby'?: React.AriaAttributes['aria-labelledby']
+  /**
+   * Custom string to use when generating the IDs of tabs and `aria-labelledby` for the panels
+   */
+  id?: string
+  /**
+   * Loading state for all counters. It displays loading animation for individual counters until all are resolved. It is needed to prevent multiple layout shift.
+   */
+  loadingCounters?: boolean
+} & SxProp
+
+export type TabProps = PropsWithChildren<{
+  /**
+   * Whether this is the selected tab
+   */
+  'aria-selected'?: boolean
+  /**
+   * Content of CounterLabel rendered after tab text label
+   */
+  counter?: number | string
+  /**
+   *  Icon rendered before the tab text label
+   */
+  icon?: FC<IconProps>
+}> &
+  SxProp
+
+export type PanelProps = Omit<BoxProps, 'as'>
+
+const TabContainerComponent = createComponent(TabContainerElement, 'tab-container')
+
+const UnderlinePanels: FC<UnderlinePanelsProps> = ({
+  'aria-label': ariaLabel,
+  'aria-labelledby': ariaLabelledBy,
+  children,
+  loadingCounters,
+  sx: sxProp = defaultSxProp,
+  ...props
+}) => {
+  const [iconsVisible, setIconsVisible] = useState(true)
+  const wrapperRef = useRef<HTMLDivElement>(null)
+  const listRef = useRef<HTMLUListElement>(null)
+  // We need to always call useId() because React Hooks must be
+  // called in the exact same order in every component render
+  const defaultId = useId()
+  const parentId = props.id ?? defaultId
```

Comment:
I think we can give the custom id as a param to the hook like `const parentId = useId(props.id)` to make it use the custom one. I find it cleaner that way but totally up to you 😊

Reply:
Oh yea I forgot we could do that. Will update.